lessons/PowerShell/http/http-portal.md

```diff
@@ -0,0 +1,305 @@
+
+    ```PowerShell
+    $name = $Request.Query.Name
+    ```
+
+    > 🔎 **Observation** - In the generated template the response was always an `[HttpStatusCode]::OK`. This means that when a clients calls the function, an HTTP status 200, is always returned. Let's make the function a bit smarter and return a `[HttpStatusCode]::BadRequest` (HTTP status 400).
+6. Add an `if` statement to the function that checks if the name value is `$null`. If the name is `$null` return a `[HttpStatusCode]::BadRequest`, otherwise return a `[HttpStatusCode]::OK`. The complete code should now look like this.
+
+    ```PowerShell
+    using namespace System.Net
+    
+    # Input bindings are passed in via param block.
+    param($Request, $TriggerMetadata)
+    
+    # Write to the Azure Functions log stream.
+    Write-Host "PowerShell HTTP trigger function processed a request."
+    
+    # Interact with query parameters or the body of the request.
+    $Name = $Request.Query.Name
+    
+    if ([string]::IsNullOrEmpty($name)) {
+        $Body = "Pass a name in the query string or in the request body for a personalized response."
+        $Result = [HttpStatusCode]::BadRequest
+    }
+    else{
+        $Body = "Hello, $Name. This HTTP triggered function executed successfully."
+        $Result = [HttpStatusCode]::OK
+    }
+    
+    # Associate values to output bindings by calling 'Push-OutputBinding'.
+    Push-OutputBinding -Name Response -Value ([HttpResponseContext]@{
+        StatusCode = $Result
+        Body = $Body
+    })
+
+    ```
+
+    Now the function has proper return values for both correct and incorrect invocations.
+
+7. Run the function, once without name value in the querystring, and once with a name value.
+
+    > ❔ **Question** - Is the outcome of both runs as expected?
+
+## 5. Changing the template for POST requests
+
+Let's change the function to also allow POST requests and test it by posting a request with JSON content in the body.
+
+### Steps
+
+1. In the left panel, click on **Integration**
+2. Click on **HTTP (Request)** under trigger
+3. Under **Selected HTTP Methods** check the **POST** option
+4. Click **Save**
+5. We need to add some logic to use the `querystring` for GET requests and use the request body for POST requests. This can be done by checking the Method property of the request (`$Request.Method`) as follows:
+
+    ```PowerShell
+    if ($Request.Method -eq "GET") {
+        $Name = $Request.Query.Name
+    }
+    elseif ($Request.Method -eq "POST"){
+        $Name = $Request.Body.Name
+    }
+    ```
+
+    >📝 **Tip** - You could also use a [Switch](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_switch?view=powershell-7.1) to reach the same goal.
+
+6. Replace the line that creates `$Name` with the `if` loop.
+
+7. Under **Code/Test** Click on **Get Function URL** and copy the URL of your function. You will need it in the next step.
```

Comment:
I suggest to add a short explanation that there are several ways to test an HTTP trigger. You can either do it in the portal (like in exercise 3) or call the endpoint from your local computer using PowerShell or a REST client.